app/controllers/admin/videos_controller.rb

```diff
@@ -0,0 +1,24 @@
+class Admin::VideosController < AdminsController
+  def new
+    @video = Video.new
+  end
+
+  def create
+    @video = Video.new(video_params)
+    category = Category.find(params[:video][:categories]) if params[:video][:categories]
+    @video.categories << category
+    if @video.save
+      flash[:success] = "You've added video #{@video.title}."
+      redirect_to new_admin_video_path
+    else
+      flash[:danger] = "Failed to create video. Please fix the problem."
```

Comment:
Whenever you are `render`ing, make sure you use `flash.now`. `flash` will persist through the next page load as well.